src/main/java/org/jitsi/jicofo/recording/Recorder.java

```diff
@@ -92,4 +98,46 @@ public void dispose()
      */
     public abstract boolean setRecording(
         String from, String token, State doRecord, String path);
-}
+
+    /**
+     * Incoming packets processing logic. Process the packet in separate thread
+     * so we do not block readers thread.
+     * <p>
+     * {@inheritDoc}
+     */
+    @Override
+    public void processPacket(Packet packet)
+    {
+        // process packet in separate thread to avoid blocking reader's thread
+        FocusBundleActivator.getSharedThreadPool()
```

Comment:
Well, we are already in a single thread situation and seeing a problem with that. The other approach is to drop and do not use sendPacketAndGetReply.